src/pointer-analysis/value_set.cpp

```diff
@@ -1611,3 +1611,22 @@ exprt value_sett::make_member(
 
   return member_expr;
 }
+
+void value_sett::erase_value_from_entry(entryt &entry, const exprt &value_to_erase)
+{
+  std::vector<object_map_dt::key_type> keys_to_erase;
+  for(const auto &key_value : entry.object_map.read())
```

Comment:
This ugliness suggests using a lower-level interface -- something like iterating over the set, translating the iterator into an exprt (i.e. to_expr), but then using the iterator to erase.

Reply:
The problem with that is that then we have to loop over `entry.object_map.write()`, which breaks sharing even if we aren't going to erase anything